dali/pipeline/executor/executor.h

```diff
@@ -84,33 +88,59 @@ class DLL_PUBLIC Executor {
   DISABLE_COPY_MOVE_ASSIGN(Executor);
 
  protected:
-  using WorkspaceBlob = struct {
-    vector<HostWorkspace> cpu_op_data;
-    vector<MixedWorkspace> mixed_op_data;
-    vector<DeviceWorkspace> gpu_op_data;
-    vector<SupportWorkspace> support_op_data;
+  struct WorkspaceBlob {
+    workspace_owner_t op_data;
+
+    void Resize(int support, int cpu, int mixed, int gpu) {
+      std::get<static_cast<int>(DALIOpType::SUPPORT)>(op_data).resize(support);
+      std::get<static_cast<int>(DALIOpType::CPU)>(op_data).resize(cpu);
+      std::get<static_cast<int>(DALIOpType::MIXED)>(op_data).resize(mixed);
+      std::get<static_cast<int>(DALIOpType::GPU)>(op_data).resize(gpu);
+    }
 
     void Clear() {
-      cpu_op_data.clear();
-      mixed_op_data.clear();
-      gpu_op_data.clear();
-      support_op_data.clear();
+      std::get<0>(op_data).clear();
```

Comment:
shouldn't those be enum values?

Reply:
They are, in some next PR, forgot to propagate it here.

Reply:
so maybe good idea to add it here before merging